extensions/s3/src/main/java/io/deephaven/extensions/s3/S3CompletableOutputStream.java

```diff
@@ -268,61 +313,102 @@ private void sendPartRequest(final OutgoingRequest request) throws IOException {
                 .partNumber(nextPartNumber)
                 .build();
         request.buffer.flip();
-        request.future = s3AsyncClient.uploadPart(uploadPartRequest,
-                AsyncRequestBody.fromByteBufferUnsafe(request.buffer));
         request.partNumber = nextPartNumber;
+        request.sdkUploadFuture = s3AsyncClient.uploadPart(uploadPartRequest,
+                AsyncRequestBody.fromByteBufferUnsafe(request.buffer))
+                .whenComplete((uploadPartResponse, throwable) -> {
+                    try {
+                        if (throwable == null) {
+                            completedParts.add(
+                                    SdkPojoConversionUtils.toCompletedPart(uploadPartResponse, request.partNumber));
```

Comment:
We don't need to hold hard ref to requests here either, and it looks like nothing else really uses `partNumber`, so I don't know if we need it in the request struct?